gadget/gadget_test.go

```diff
@@ -361,6 +368,54 @@ func (s *gadgetYamlTestSuite) TestReadGadgetYamlOnClassicOnylDefaultsIsValid(c *
 	})
 }
 
+func (s *gadgetYamlTestSuite) TestFlatten(c *C) {
+	cfg := map[string]interface{}{
+		"foo":         "bar",
+		"some.option": true,
+		"sub": map[string]interface{}{
+			"option1": true,
+			"option2": map[string]interface{}{
+				"deep": "2",
+			},
+		},
+	}
+	out := map[string]interface{}{}
+	gadget.Flatten("", cfg, out)
+	c.Check(out, DeepEquals, map[string]interface{}{
+		"foo":              "bar",
+		"some.option":      true,
+		"sub.option1":      true,
+		"sub.option2.deep": "2",
+	})
+}
+
+func (s *gadgetYamlTestSuite) TestCoreConfigDefaults(c *C) {
+	err := ioutil.WriteFile(s.gadgetYamlPath, mockClassicGadgetCoreDefaultsYaml, 0644)
+	c.Assert(err, IsNil)
+
+	ginfo, err := gadget.ReadInfo(s.dir, &modelConstraints{classic: true})
+	c.Assert(err, IsNil)
+	defaults := gadget.SystemDefaults(ginfo.Defaults)
+	c.Check(defaults, DeepEquals, map[string]interface{}{
+		"ssh.disable": true,
+	})
+
+	yaml := string(mockClassicGadgetCoreDefaultsYaml) + `
+  system:
+    something: true
```

Comment:
it would be good if this had one level more of nesting,  otherwise there really no connection between the behavior here and flatten which is fully tested separately

Reply:
Done.